MapboxMobileEvents/MMEEvent.h

```diff
@@ -1,42 +1,172 @@
-+ (instancetype)locationEventWithAttributes:(NSDictionary *)attributes instanceIdentifer:(NSString *)instanceIdentifer commonEventData:(MMECommonEventData *)commonEventData;
+
+/*! @brief visitEventWithAttributes:
+    @param attributes attrs
+    @return event
+*/
 + (instancetype)visitEventWithAttributes:(NSDictionary *)attributes;
-+ (instancetype)mapLoadEventWithDateString:(NSString *)dateString commonEventData:(MMECommonEventData *)commonEventData;
-+ (instancetype)mapTapEventWithDateString:(NSString *)dateString attributes:(NSDictionary *)attributes;
-+ (instancetype)mapDragEndEventWithDateString:(NSString *)dateString attributes:(NSDictionary *)attributes;
-+ (instancetype)mapOfflineDownloadStartEventWithDateString:(NSString *)dateString attributes:(NSDictionary *)attributes;
-+ (instancetype)mapOfflineDownloadEndEventWithDateString:(NSString *)dateString attributes:(NSDictionary *)attributes;
-+ (instancetype)navigationEventWithName:(NSString *)name attributes:(NSDictionary *)attributes;
-+ (instancetype)visionEventWithName:(NSString *)name attributes:(NSDictionary *)attributes;
+
+#pragma mark - Crash Events
+
+/*! @brief crashEventReporting:error:
+    @param eventsError error to report
+    @param createError pointer to an error creating the report
+    @return event
+*/
++ (instancetype)crashEventReporting:(NSError *)eventsError error:(NSError **)createError;
+
+#pragma mark - Debug Devents
+
+/*! @brief debugEventWithAttributes: debug logging event with attributes provided
+    @param attributes attrs
+    @return event
+*/
 + (instancetype)debugEventWithAttributes:(NSDictionary *)attributes;
-+ (instancetype)debugEventWithError:(NSError*) error;
-+ (instancetype)debugEventWithException:(NSException*) except;
-+ (instancetype)searchEventWithName:(NSString *)name attributes:(NSDictionary *)attributes;
-+ (instancetype)carplayEventWithName:(NSString *)name attributes:(NSDictionary *)attributes;
-+ (instancetype)eventWithDateString:(NSString *)dateString name:(NSString *)name attributes:(NSDictionary *)attributes;
 
+/*! @brief debugEventWithError: debug logging event with the error provided
+    @param error error
+    @return event
+*/
++ (instancetype)debugEventWithError:(NSError *)error;
+
+/*! @brief debugEventWithException: debug logging event the the exception provided
+    @param except exception
+    @return event
+*/
++ (instancetype)debugEventWithException:(NSException *)except;
+
+#pragma mark - Deprecated
+
+#pragma mark - Deprecated (MMECommonEventData)
+
+/*! @brief deprecated in MabboxMobileEvents 1.0.0 or later
+    @note please use eventWithAttributes:error:
+*/
++ (instancetype)locationEventWithAttributes:(NSDictionary *)attributes instanceIdentifer:(NSString *)instanceIdentifer commonEventData:(MMECommonEventData *)commonEventData
+    MME_DEPRECATED_GOTO("use eventWithAttributes:error:", "-eventWithAttributes:error:");
```

Comment:
When both a deprecation message and a replacement are specified, the warning within Xcode is duplicative. In the map and navigation SDKs, I’ve been specifying the empty string as the message when there’s a replacement.